packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2937,6 +2950,18 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   double get _devicePixelRatio => MediaQuery.of(context).devicePixelRatio;
 
+  /// Indicates that the user has requested to apply [TextEditingDelta]s to its
+  /// current text editing state with [deltas].
+  void userUpdateTextEditingValueWithDeltas(List<TextEditingDelta> deltas, SelectionChangedCause? cause) {
+    TextEditingValue newValue = _value;
+
+    for (final TextEditingDelta delta in deltas) {
+      newValue = delta.apply(newValue);
+    }
+
+    userUpdateTextEditingValue(newValue, cause);
```

Comment:
Would it be possible to eventually make this private so everything has to go through userUpdateTextEditingValueWithDeltas?

Reply:
That's possible, though it is part of `TextSelectionDelegate`. Should `userUpdateTextEditingValueWithDeltas` be a part of `TextSelectionDelegate` as well. I was thinking that might be what we need to do.

Reply:
This wouldn't help too much on my demo since it doesn't extend `EditableText`, and has to implement `copy/cut/paste` on its own. This would be helpful for the use case of extending `EditableText` like in #103661. 

Though thinking about it I wonder if the solution here is to override the text editing actions like `ReplaceTextIntent` and `DeleteTextIntent` (make a new one) and `UpdateSelectionIntent` above the `EditableText`. Though there are some places where the value isn't set by an intent -> actions mapping, sometimes we directly call `_replaceText(intent)`. Overall I still think it is a good idea to move towards deltas.